Book-A-Desk.Api.Tests/BookingsTests.fs

```diff
@@ -70,13 +75,113 @@ let ``GIVEN A Book-A-Desk server, WHEN booking a desk, THEN a desk is booked`` (
 
 [<Fact>]
 let ``GIVEN A Book-A-Desk server, WHEN booking a desk, THEN an email notification must be sent`` () = async {
-    emailWasSent <- false    
+    emailWasSent <- false
     let mockApiDependencyFactory = ApiDependencyFactory.provide mockProvideEventStore mockReservationCommandFactory mockGetOffices mockEmailNotification
     use httpClient = TestServer.createAndRun mockApiDependencyFactory
     
     let serializedBooking = JsonConvert.SerializeObject(booking)
 
-    let! result = HttpRequest.postAsync httpClient $"http://localhost:/bookings" serializedBooking
+    let! result = HttpRequest.postAsync httpClient url serializedBooking
     
     Assert.True(emailWasSent)
 }
+
+[<Fact>]
+let ``GIVEN an invalid reservation details WHEN booking a desk THEN it returns 400 And Reservation Error Title and Details And no notification by email is sent`` () = async {
+    emailWasSent <- false
+    let mockApiDependencyFactory = ApiDependencyFactory.provide mockProvideEventStore mockReservationCommandFactory mockGetOffices mockEmailNotification
+    use httpClient = TestServer.createAndRun mockApiDependencyFactory
+    
+    let bookingInvalidEmail  =
+        {
+            booking with User = { Email = "invalidEmail" }
+        } : InputBooking
+
+    let expectedTitle = "Invalid Email Address"
+    let expectedDetails = "The e-mail address is invalid."
+    
+    let serializedBooking = JsonConvert.SerializeObject(bookingInvalidEmail)
+
+    let postRequest = new HttpRequestMessage(HttpMethod.Post, url)
+    let content = new StringContent(serializedBooking, Encoding.UTF8, "application/json")
+    postRequest.Content <- content
+
+    httpClient.SendAsync postRequest |> Async.AwaitTask |> Async.RunSynchronously
```

Comment:
Suggestion: this code to send post request is duplicated 3 times in this class and could be extracted to a function (with status code parameter) for better readability and avoids code duplication.

Reply:
Good idea, Not only I've created a separate function for this, but also I renamed the existing ones in HttpRequests as they were returning just the content and checking that the requests are returning 200 Success. However, in error tests we need to review the Status and the serialised response.

Reply:
Fixed now!